File: lessons/rogerfm.net/crackme/02/keygen.py
#!/usr/bin/env python
# encoding:utf-8

import sys

def get_password(username):
    password = 0
    i = 0
    j = 45
    while True:
        if i >= len(username):
            break
        password += j * ord(username[i])
        i += 1
        j += 1
    print "Username : [ %s ]\nPassword : [ %s ]" % (username, password)

def show_help():
    print "Usage : \n\tpython %s [username]" % sys.argv[0]

def main():
    if len(sys.argv) != 2:
        show_help()
        exit(1)
    get_password(sys.argv[1])

if __name__ == "__main__":
    main()

File: lessons/rogerfm.net/crackme/02/keygen.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void get_password(char *username);

int main(int argc, char *argv[]){
	if (argc != 2){
		printf("Usage : \n\t%s [username]\n", argv[0]);
		exit(1);
	}

	char *username = argv[1];
	get_password(username);

	return 0;
}

void get_password(char *username){
	int length = strlen(username);
	int i;
	int j;
	int password = 0;
	for(i = 0, j = 45; i < length; i++, j++){
		password += username[i] * j;
	}
	printf("Username : [ %s ]\nPassword : [ %d ]\n", username, password);
}
